### ISSUE TITLE
LineTrace fails to collide with tiles at a long distance

### ISSUE THREAD

**Issue by putimj** [CesiumGS/cesium-unreal#1078]
When the camera is far from the ground, LineTrace fails to collide with tiles. This issue is easier to observe when there are several different levels of tiles in the view, such as level 9 in the middle and level 8 next to it, in this case,In this case, level 9 tiles can collide, but adjacent level 8 tiles will fail.
Any help would be appreciated.

**Comment by kring** [comment]
When you do a line trace, you set a start and end point. Is there any chance your end point is simply too close, and occurs before the terrain? Can you share the code or Blueprint you're using to do the line trace?

**Comment by putimj** [comment]
Here's my test code in ACesiumGeoreference::Tick:
```
  UWorld* world = this->GetWorld();
  if (world == NULL)
      return;
  APlayerController* controller = world->GetFirstPlayerController();
  if (controller == NULL)
      return;
  FVector2D DIR;
  FVector worldpos;
  FVector worlddirect;

  FVector2D mousepos;
  controller->GetMousePosition(mousepos.X, mousepos.Y);
  UGameplayStatics::DeprojectScreenToWorld(controller, FVector2D(mousepos.X, mousepos.Y), worldpos, worlddirect);

  FVector viewLoc = worldpos;
  FVector viewDir = worlddirect;
  const double r = 63710000000.0;
  FVector lineEnd = viewLoc + viewDir * r;

  FCollisionQueryParams CollisionParams(LineTraceSingleName);
  FCollisionObjectQueryParams ObjectParams =
      FCollisionObjectQueryParams(ECC_WorldStatic);
  ObjectParams.AddObjectTypesToQuery(ECC_WorldDynamic);
  ObjectParams.AddObjectTypesToQuery(ECC_Visibility);
  FHitResult HitResult;
  if (world->LineTraceSingleByObjectType(
      HitResult,
      viewLoc,
      lineEnd,
      ObjectParams,
      CollisionParams))
  {
      FVector hitLocation = HitResult.Location;
      DrawDebugPoint(world, hitLocation, 10.0, FColor(255, 0, 0), false);
  }
```

**Comment by putimj** [comment]
![image](https://user-images.githubusercontent.com/24795458/234827757-ee28c994-18d9-41f4-b7c9-43de1244ca9a.png)
It only collided in the red region.
My engine version is 5.1.
@kring

**Comment by kring** [comment]
Your code looks ok from what I can see. But there's nothing particularly special about Cesium meshes once they're loaded, so I don't know what on our end would cause this. Perhaps it's a bug or limitation in `LineTraceSingleByObjectType`? Which UE version are you using?

**Comment by kring** [comment]
Is that Cesium World Terrain you're attempting to collide with, or something else?

**Comment by putimj** [comment]
Yes, I'm  attempting to collide with Cesium World Terrain to display the longitude and latitude of the point where the mouse is located. And I have tested it on several computers(Windows11), all of which are like this.

**Comment by kring** [comment]
This was fixed in v1.28.0.
